Tryouts/Core/Abstractions/ISubscriptionClient.cs

```diff
@@ -1,27 +1,11 @@
-﻿/// ********************************************************************************************************
-///
-/// Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License").
-/// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
-/// See the NOTICE file distributed with this work for additional information regarding copyright ownership.
-/// Unless required by applicable law or agreed to in writing, software distributed under the License
-/// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-/// See the License for the specific language governing permissions and limitations under the License.
-/// 
-/// ********************************************************************************************************
-
-using Google.Protobuf;
-using Subscriptions;
-
-namespace MorganStanley.ComposeUI.Tryouts.Core.Abstractions
+﻿namespace MorganStanley.ComposeUI.Tryouts.Core.Abstractions
 {
     public interface ISubscriptionClient
     {
         Task Connect(string host, int port);
 
-        Task Publish<TMessage>(Topic topic, TMessage msg)
-             where TMessage : IMessage;
+        Task Publish(string topic, string msg);
 
-        IAsyncEnumerable<TMessage> Subscribe<TMessage>(Topic topic)
-            where TMessage : IMessage, new();
+        IObservable<CommunicationsMessage> Subscribe(string topic);
```

Comment:
I'm not sure 'Subscribe' would be the best name for this method since it wouldn't actually subscribe to anything - you'd need to call Subscribe() on the returned IObservable. @BalassaMarton suggested that this method could be called 'Topic' or something similar - that would make the code more readable.

On the other hand I'm a bit ambivalent about IObservable vs IAsyncEnumerable. I understand that one can be converted to the other and vice-versa, but looking at the current implementation of this interface it'd feel more _natural_ to simply return IAsyncEnumerable instead of IObservable.

Reply:
I think we should go by what the clients need, rather than by implementation. Observable is smth people are comfortable - it has been around for longer and there are plenty examples. IAsyncEnumerable - I am not even sure if it is possible to use Linq operators on them without converting them first to IObservable. So I think in order to make it easier for the clients - it is better to keep them as IObservable.

Reply:
I agree renaming this method to Topic, or something similar, currently this results in MessageRouter.Subscribe("topic").Subscrive(observer);

For me, IObservable feels more natural, but I guess it's a matter of taste and the use-case, I don't mind either too much.